### ISSUE TITLE
Is save_stats_to_file writing to the correct path?

### ISSUE THREAD

**Issue by jasonthomas** [mozilla/addons-server#7991]
I noticed on prod that save_stats_to_file is putting stats data into `/mnt/efs/addons.mozilla.org/shared_storage/uploads/<addon_id>/`. I think this data should be put into sub directory instead, maybe `/mnt/efs/addons.mozilla.org/shared_storage/uploads/statistics/<addon_id>/` so that the uploads directory is easier to manage.

https://github.com/mozilla/addons-server/blob/318d0dc4e35c1e28a3e9c41e76d66e518399ce9d/src/olympia/stats/management/commands/__init__.py#L98

**Comment by EnTeQuAk** [comment]
> Or maybe we can get rid of save_stats_to_file? I remember the idea was to save stats to file and have them accessible via the stats api but I am not sure how much we care about that now. AFAIK we have not been pruning stats data from the database.

I also don't see any usage data on `ArchiveListView` and `ArchiveView` in NewRelic (which are used to download statistic data), docs are in https://addons-server.readthedocs.io/en/latest/topics/api/stats.html

Do you see anywhere else some usage data of these APIs?

Given that it's in apiv3 and we want to make apiv3 stable sometime soon or rather now we should maybe usefulness of these APIs and saving of stats data.

This got implemented in https://github.com/mozilla/addons-server/issues/1414 as part of https://github.com/mozilla/addons-server/issues/1395 which is still open but might be obsolete

cc @eviljeff as any other voice is pretty much on PTO right now :)

**Comment by eviljeff** [comment]
From a quick look at AMO stats dashboard for ABP we _do_ prevent older stats from being retrieved via the website currently, so we can't just remove the APIs.  I'm assuming it's a block in the view code somewhere we can easily remove though.

I've no idea about the original justification - that old stats were a huge problem - I wasn't working on AMO full time then.  IMO either we've got to allow old stats to be downloaded via some endpoint or via the website 🤷‍♂️

**Comment by jasonthomas** [comment]
I believe `save_stats_to_file` and archiving stats were done to mitigate performance issues in AWS since we thought that the large stats table within the database was the culprit. It ended up not being the real issue.

Even though the stats table continue to grow I think we could just continue to maintain them in the database and have the APIs reference the data directly. We also spoke in the past about moving these stats table into it's own database/database cluster but not sure if we really need to do that today.  Including  @bqbn @philipcristiano for comment.

**Comment by bqbn** [comment]
> I think this data should be put into sub directory instead, maybe /mnt/efs/addons.mozilla.org/shared_storage/uploads/statistics/<addon_id>/ so that the uploads directory is easier to manage.

Second that! A directory with a lot files in it has always been an issue for us. Every time we can have a nested directory to store files, we should probably do it.

> We also spoke in the past about moving these stats table into it's own database/database cluster but not sure if we really need to do that today.

I vote for having the stats in a separate database cluster if possible. Daily stats indexing causes a lot IOs to the database during which it may have impacted users browsing the website.

**Comment by EnTeQuAk** [comment]
I solved this in https://github.com/mozilla/addons-server/pull/8091 but linked to the wrong issue. We decided to step back from statistics archiving and are not saving any more data to EFS. We also removed the respective views.

Let's give this code a week to land properly and see if there were any actual users we didn't notice in DataDog or NewRelic and if not let's get rid of the archived stats in EFS. We still have the data in the database anyway.
